migrations/2019-10-30-155718_schema/up.sql

```diff
@@ -1,14 +1,25 @@
-CREATE TABLE users (
-    id       SERIAL PRIMARY KEY,
-    username VARCHAR(255) NOT NULL UNIQUE,
-    password VARCHAR(255) NOT NULL,
-    admin    BOOLEAN NOT NULL DEFAULT false
+CREATE TABLE users
+(
+    id              SERIAL PRIMARY KEY,
+    username        VARCHAR(255) NOT NULL UNIQUE,
+    hashed_password VARCHAR(255) NOT NULL,
+    admin           BOOLEAN      NOT NULL DEFAULT false,
+    firstname       VARCHAR(255) NOT NULL,
+    lastname        VARCHAR(255) NOT NULL,
+    email           VARCHAR(255) NOT NULL UNIQUE,
+    ssh_key         TEXT,
+    last_seen       TIMESTAMP    NOT NULL,
```

Comment:
```suggestion
    last_login      TIMESTAMP    NOT NULL,
```
A 'last_seen' implies that on every request we save the current timestamp of the user performing the requests. This doesn't seem like a good idea. This would not scale well, would put a lot of avoidable load on the database server and would render the system completely unusable in the case of a read-only database.

Reply:
This is linked to how long a token will be valid because we want to have a bit of an idea of how much users are active.

Are refresh token used? We could let those be long and the normal tokens shorter. And on every refresh token usage update the last_login/seen field. It would be somewhat in between...?

What do you think @redfast00 ?

Reply:
The current session is cookie-based without refresh tokens, and I would make those sessions valid for only a short period (a day).